packages/txservice/src/chainreader.ts

```diff
@@ -6,6 +6,9 @@ import { ReadTransaction } from "./types";
 import { ChainRpcProvider } from "./provider";
 import { ConfigurationError, ProviderNotConfigured } from "./error";
 
+// TODO: I do not like that this is generally a passthrough class now - all it handles is the mapping. We should
+// probably just expose a provider getter method and have the consumer call that to access the target ChainRpcProvider
+// directly.
```

Comment:
would you want to include this in this PR or a separate one?

Reply:
separate for now. ran some ideas by rahul, we'll catch up on this in standup tmrw